statediff/service.go

```diff
@@ -147,32 +125,49 @@ func (sds *Service) Loop(chainEventCh chan core.ChainEvent) {
 	}
 }
 
-// processStateDiff method builds the state diff payload from the current and parent block before sending it to listening subscriptions
-func (sds *Service) processStateDiff(currentBlock, parentBlock *types.Block) error {
-	stateDiff, err := sds.Builder.BuildStateDiff(parentBlock.Root(), currentBlock.Root(), currentBlock.Number(), currentBlock.Hash())
-	if err != nil {
-		return err
+func (sds *Service) processStateChanges(stateChangeEvent core.StateChangeEvent) error {
+	var accountDiffs []AccountDiff
+	modifiedAccounts := stateChangeEvent.StateChanges.ModifiedAccounts
+	for addr, modifiedAccount := range modifiedAccounts {
+		accountBytes, err := rlp.EncodeToBytes(modifiedAccount.Account)
+		if err != nil {
+			return err
+		}
+
+		var storageDiffs []StorageDiff
+		for k, v := range modifiedAccount.Storage {
+			diff := StorageDiff{
+				Key:   k[:],
+				Value: v[:],
+			}
+			storageDiffs = append(storageDiffs, diff)
+		}
+
+		address := addr
```

Comment:
What's the purpose of this assignment?

Reply:
Great question. When I didn't assign `addr` to a new variable, I was seeing some state weirdness, where it seemed like as we iterated through the `modifiedAccounts`, the address would be overwritten in the returned `accountDiffs` collection. So they'd all have the same address for some reason. I haven't fully figured it out yet, but I'll probably spend some more time digging in to see if there's a better solution.